File: back-end/src/main/java/es/ull/project/domain/valueobject/ExportFile.java
package es.ull.project.domain.valueobject;

import java.util.Objects;

import es.ull.project.domain.entity.WorkFlow;

/**
 * @brief Represents an exported file(JSON) with name and binary data.
 */
public class ExportFile {

    private static final String ERROR_DATA_NOT_DEFINED = "File data not defined";
    private final WorkFlow data;

    /**
     * Constructor for ExportFile.
     * 
     * @param fileName
     * @param data
     */
    public ExportFile(WorkFlow data) {
        this.validateData(data);
        this.data = data;
    }

    private void validateData(WorkFlow data) {
        if (data == null) {
            throw new IllegalArgumentException(ERROR_DATA_NOT_DEFINED);
        }
    }

    public WorkFlow getData() {
        return this.data;
    }

    public ExportFile setData(WorkFlow data) {
        return new ExportFile(data);
    }

    @Override
    public boolean equals(Object otherObject) {
        if (this == otherObject) {
            return true;
        }
        if (!(otherObject instanceof ExportFile)) {
            return false;
        }
        final ExportFile that = (ExportFile) otherObject;
        return Objects.equals(data, that.data);
    }

    @Override
    public int hashCode() {
        return Objects.hash(data);
    }

    @Override
    public String toString() {
        return String.format(
                "ExportFile={data=%s}",
                this.data);
    }
}

File: front-end/src/domain/value-object/export-file.ts
import { WorkFlow } from "../entity/workflow";

/**
 * Represents an exported workflow file with a name and workflow data.
 */
export class ExportFile {

    private static readonly ERROR_FLOW_NOT_DEFINED = 'File data not defined';

    private readonly workflow: WorkFlow;

    constructor(workflow: WorkFlow) {
        this.validateFlow(workflow);
        this.workflow = workflow;
    }

    private validateFlow(workflow: WorkFlow): void {
        if (!workflow) {
            throw new Error(ExportFile.ERROR_FLOW_NOT_DEFINED);
        }
    }

    public getWorkFlow(): WorkFlow {
        return this.workflow;
    }

    public setWorkFlow(workflow: WorkFlow): ExportFile {
        return new ExportFile(workflow);
    }


    public toString(): string {
        return `ExportFile={workflowName='${this.workflow.getName().getName()}'}`;
    }

    public equals(other: ExportFile): boolean {
        return other instanceof ExportFile && this.workflow === other.workflow;
    }
}
